sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/ProjectSettings.java

```diff
@@ -36,6 +37,9 @@
   private final DefaultAnalysisMode mode;
   private final Map<String, String> properties = new HashMap<>();
 
+  /*
+   * Using ProjectReactor instead of InputModuleHierarchy to avoid cyclic dependency
+   */
   public ProjectSettings(ProjectReactor reactor, GlobalSettings globalSettings, ProjectRepositories projectRepositories, DefaultAnalysisMode mode) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "ProjectReactor"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)